adafruit_requests.py

```diff
@@ -440,7 +522,7 @@ def _free_sockets(self) -> None:
 
     def _get_socket(
         self, host: str, port: int, proto: str, *, timeout: float = 1
-    ) -> SocketType:
+    ) -> CircuitPythonSocketType:
```

Comment:
This change was needed because the `recv_into()` method is called on the socket returned by this `_get_socket()` method in the `request()` method, and not every `SocketType` provides a `recv_into()` method (specifically `LegacyCircuitPythonSocketType` does not). `CircuitPythonSocketType` meets that requirement while also being a valid `SocketType` to support the socket being passed to other methods which require a `SocketType` socket.